### PR TITLE
Use k8s types and validation code with the TriggerResourceTemplate type

### PR DESCRIPTION
**Pull request by gabemontero**
# Changes

This PR introduces some k8s alternatives to 
- `TriggerResourceTemplate` raw payload  (see https://github.com/tektoncd/triggers/pull/51#issuecomment-520876087 through https://github.com/tektoncd/triggers/pull/51#issuecomment-520884302 from last year, followed by https://github.com/tektoncd/triggers/pull/51#issuecomment-582103292 through https://github.com/tektoncd/triggers/pull/51#issuecomment-582159404 from a few days ago for history on the path to doing this)
- then while I was in here, I saw with the introduction of the dynamic clients the opportunity to use the k8s runtime scheme/decoder approach to validating the `TriggerResourceTemplate`

I delivered each in their own commit to facilitate the possibility of dropping one, but not both, of these changes.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [/] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [/] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/triggers/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

@dibyom @vdemeester @vtereso FYI / PTAL

thanks

### CODE REVIEW

**Comment by gabemontero** [comment]
My initial guesses at filling out the CLA form were unsuccessful.  I've sent a query out to the other RH tekton devs to find out what the secret sauce is.

File: pkg/apis/triggers/v1alpha1/trigger_template_types.go [hunk 1]
```diff
@@ -48,7 +46,7 @@ type TriggerTemplateSpec struct {
 
 // TriggerResourceTemplate describes a resource to create
 type TriggerResourceTemplate struct {
-	json.RawMessage `json:",inline"`
+	runtime.RawExtension `json:",inline"`
```
Comment (vdemeester) [inline pkg/apis/triggers/v1alpha1/trigger_template_types.go]: :heart_eyes_cat:

**Comment by gabemontero** [comment]
OK I've been added to the RH/Google CLA

@googlebot I signed it!

**Comment by gabemontero** [comment]
I also have a commit to address the `TODO` at https://github.com/tektoncd/triggers/blob/master/pkg/apis/triggers/v1alpha1/trigger_template_validation.go#L34 and validate the `ObjectMetadata` for `TriggerTemplate` that I can push upif you all would like.

**Comment by dibyom** [comment]
> I also have a commit to address the TODO at https://github.com/tektoncd/triggers/blob/master/pkg/apis/triggers/v1alpha1/trigger_template_validation.go#L34 and validate the ObjectMetadata for TriggerTemplate that I can push up if you all would like.

Sure, this or a new PR for that are both fine :smile:

**Comment by gabemontero** [comment]
> > I also have a commit to address the TODO at https://github.com/tektoncd/triggers/blob/master/pkg/apis/triggers/v1alpha1/trigger_template_validation.go#L34 and validate the ObjectMetadata for TriggerTemplate that I can push up if you all would like.
> 
> Sure, this or a new PR for that are both fine smile

OK cool ... if anybody else requests changes on the existing stuff I'll push it up with those changes.  Otherwise I'll just pull the commit into a new PR.

thanks

**Comment by gabemontero** [comment]
bump @wlynch @ncskier @dibyom 

You all happy with the changes?

I do plan on joining the triggers sync call today if discussing this there is warranted in any of your opinions.

**Comment by wlynch** [comment]
/lgtm

This looks great! :D 

Thanks for the change!

**Comment by gabemontero** [comment]
cool / thanks / glad to help @wlynch 

I'll pull in that ObjectMetadata TODO into a new PR.

**Pull request merged**